concepts/pipeline/event_pipeline.md

```diff
@@ -0,0 +1,127 @@
+    {"prot": "https", "src": "script"}],
+  // ...
+]
+```
+
+Each event is of the form:
+
+```javascript
+[timestamp, category, method, object, value, extra]
+```
+
+Where the individual fields are:
+
+- `timestamp`: `Number`, positive integer. This is the time in ms when the event was recorded, relative to the main process start time.
+- `category`: `String`, identifier. The category is a group name for events and helps to avoid name conflicts.
+- `method`: `String`, identifier. This describes the type of event that occurred, e.g. `click`, `keydown` or `focus`.
+- `object`: `String`, identifier. This is the object the event occurred on, e.g. `reload_button` or `urlbar`.
+- `value`: `String`, optional, may be null. This is a user defined value, providing context for the event.
+- `extra`: `Object`, optional, may be null. This is an object of the form `{"key": "value", ...}`, both keys and values need to be strings. This is used for events when additional richer context is needed.
+
+
+# Event data collection
+
+## Firefox event collection
+
+To collect this event data in Firefox there are different APIs in Firefox, all addressing different
+use cases:
+- The [*Telemetry event API*](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/events.html)
+  allows easy recording of events from Firefox code.
+- The [*dynamic event API*](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/events.html#registerevents)
+  allows code from Mozilla addons to record new events into Telemetry without shipping Firefox
+  code.
+- The *Telemetry extension API* ([work in progress](https://bugzilla.mozilla.org/show_bug.cgi?id=1280234))
+  will allow Mozilla extensions to record new events into Telemetry.
+- The [*Hybrid-content API*](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/hybrid-content.html)
+  allows specific white-listed Mozilla content code to record new events into Telemetry.
+
+For all these APIs, events will get sent to the pipeline through the
+[main ping](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/data/main-ping.html),
+with a hard limit of 500 events per ping.
+In the future, Firefox events will be sent through a separate *events ping*, removing the hard limit.
+As of Firefox 61, all events recorded through these APIs are [automatically counted in scalars](https://bugzilla.mozilla.org/show_bug.cgi?id=1440673).
+
+Finally, [*custom pings*](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/custom-pings.html)
+can follow the event data format and potentially connect to the existing tooling with some integration work.
+
+## Mobile event collection
+
+On mobile, event data is currently collected through the [`focus-events` ping](https://github.com/mozilla-mobile/focus-ios/wiki/Event-Tracking-with-Mozilla%27s-Telemetry-Service#event-ping),
```

Comment:
I would change this to the `mobile-events` ping, which is the one we want people to be using.

Reply:
Do we support this ping yet?
We probably should clearly call out what we have and what is just coming.

Reply:
For over 4 months now :)

https://github.com/mozilla-services/mozilla-pipeline-schemas/tree/dev/schemas/telemetry/mobile-event